src/helm/benchmark/contamination/document_reading_processor.py

```diff
@@ -0,0 +1,60 @@
+import json
+
+from dataclasses import dataclass
+from typing import Generator
+
+
+@dataclass(frozen=True, eq=False)
+class DocumentReadingProcessor:
+    """
+    TODO: This is probably a bad class name.
```

Comment:
I don't think you need a dataclass for this.  You can just have a function that takes the `file_path` and `file_format` and returns a `Iterable[str]`